### PR TITLE
Add basic tests

### LINKED ISSUE #23: Setting up continuous integration

### PR DESCRIPTION
**Pull request by miguel-s**
With this PR I want to kickstart our testing efforts for `surface_site`.

#### CI
Setup CI with Github Actions (setup is identical to https://github.com/surface-ui/surface).
Closes #23.

#### Tests
One test per top-level page to ensure everything loads ok and we don't accidentally break anything:
- [x] Home
- [x] Getting started
- [x] Documentation
  - [x] Components Basics
  - [x] Properties
  - [x] Data
  - [x] Events
  - [x] Slots
  - [x] State Management
  - [x] Contexts
  - [x] Testing
  - [x] JS Interoperability
- [x] Built-in Components
  - [x] Link
  - [x] LivePatch
  - [x] LiveRedirect
  - [x] Form
  - [x] Field
  - [x] FieldContext
  - [x] Label
  - [x] Input Controls
  - [x] TextArea
  - [x] Select
  - [x] ErrorTag
  - [x] Markdown
  - [x] Raw
- [x] UI Components (WIP)
  - [x] Button
  - [x] Table
  - [x] Tabs

#### Fix bugs
1. Add missing `id` attribute to container when using `phx-update`
2. Add missing dependency `phoenix_ecto`, necessary for the `Contexts` page

### CODE REVIEW

**Comment by miguel-s** [comment]
@msaraiva second that is stuck at `Run mix format --check-formatted`. Do you remember why we added `check_formatted: true` in that case?

**Comment by msaraiva** [comment]
@miguel-s we only run the formatter for the latest stable version as the formatting rules may change between versions. 

I believe fix the problem by moving the task to after the deps are installed. Like I did here https://github.com/surface-ui/surface_markdown/commit/6710419933491e0986f2877e0c8acc243f002188#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f

File: lib/surface_site_web/live/contexts/example01.ex [hunk 1]
```diff
@@ -4,8 +4,8 @@ defmodule SurfaceSiteWeb.Contexts.Example01 do
     import Ecto.Changeset
 
     schema "users" do
-      field :email, :string
-      field :name, :string
+      field(:email, :string)
+      field(:name, :string)
```
Comment (msaraiva) [inline lib/surface_site_web/live/contexts/example01.ex:8]: Was that the formatter? If so, you could try to add `:ecto` to https://github.com/surface-ui/surface_site/blob/3dca3829635d684f530c2be8fdf98b69340a9aea/.formatter.exs#L2

**Review by msaraiva — APPROVED** [review]
@miguel-s thanks for adding those tests ❤️

I left just two tiny comments for your consideration. Feel free to merge it after addressing (or refusing) them :)

File: lib/surface_site_web/live/contexts/example01.ex (hunk 1 above)
Comment (miguel-s) [reply lib/surface_site_web/live/contexts/example01.ex]: Yes will do that, didn't even cross my mind to check the formatter config 😅

File: lib/surface_site_web/components/markdown.ex [hunk 2]
```diff
@@ -29,8 +29,16 @@ defmodule SurfaceSiteWeb.Components.Markdown do
...
   # Update <pre> for code mermaid
-  def post_processor({"pre", attrs, [{"code", [{"class", "mermaid language-mermaid"}], _, _}], meta}) do
-    attrs = [{"class", "language-mermaid"}, {"phx-update", "ignore"} | attrs]
+  def post_processor(
+        {"pre", attrs, [{"code", [{"class", "mermaid language-mermaid"}], _, _}], meta}
+      ) do
+    id = :erlang.unique_integer([:positive])
+
+    attrs = [
+      {"id", to_string(id)},
```
[the remark this answers is not in the document]
Comment (miguel-s) [reply lib/surface_site_web/components/markdown.ex]: Yes we should, should have checked how ids are handled already. Will add prefixes to all new ids.

**Pull request merged**